File: backend/IntexAPI/IntexAPI/Models/ApiDtos.cs
namespace IntexAPI.Models;

// --- Public impact ---
public record PublicImpactSummaryDto(
    int ActiveResidentsCount,
    int SafehouseCount,
    double TotalDonationsAllTime,
    int TotalGirlsServed,
    IReadOnlyList<PublicImpactSnapshotSummaryDto> LatestPublishedSnapshots);

public record PublicImpactSnapshotSummaryDto(
    int SnapshotId,
    DateOnly? SnapshotDate,
    string? Headline,
    string? SummaryText);

// --- Admin dashboard ---
public record AdminDashboardDto(
    int TotalActiveResidents,
    IReadOnlyList<ResidentsBySafehouseDto> ActiveResidentsBySafehouse,
    IReadOnlyList<RecentDonationDto> RecentDonations,
    IReadOnlyList<UpcomingCaseConferenceDto> UpcomingCaseConferences,
    IReadOnlyList<LatestSafehouseProgressDto> LatestMonthlyProgressBySafehouse,
    int TotalSupporters,
    int SupportersDonatedThisMonth);

public record ResidentsBySafehouseDto(int SafehouseId, string? SafehouseName, string? SafehouseCode, int ActiveResidentCount);

public record RecentDonationDto(
    int DonationId,
    DateOnly? DonationDate,
    string? DonationType,
    double? Amount,
    string? CurrencyCode,
    int? SupporterId,
    string? SupporterDisplayName);

public record UpcomingCaseConferenceDto(
    int PlanId,
    int? ResidentId,
    string? ResidentCaseNo,
    DateOnly? CaseConferenceDate,
    string? PlanCategory,
    string? Status);

public record LatestSafehouseProgressDto(
    int? SafehouseId,
    string? SafehouseName,
    DateOnly? MonthStart,
    int? ActiveResidents,
    double? AvgEducationProgress,
    double? AvgHealthScore,
    int? IncidentCount);

// --- Analytics ---
public record DonationTrendPointDto(int Year, int Month, double TotalAmount, int DonationCount);

public record ReintegrationStatusCountDto(string? Status, int Count);

public record SafehousePerformanceDto(
    int SafehouseId,
    string? SafehouseName,
    DateOnly? MonthStart,
    int? ActiveResidents,
    double? AvgEducationProgress,
    double? AvgHealthScore,
    int? ProcessRecordingCount,
    int? HomeVisitationCount,
    int? IncidentCount);

public record OutcomeTrendPointDto(
    DateOnly MonthStart,
    double? AvgEducationProgress,
    double? AvgHealthScore,
    int ActiveResidentsSum);

File: frontend/src/types/apiDtos.ts
/** Shapes returned by aggregate endpoints (camelCase JSON from ASP.NET). */

export interface PublicImpactSnapshotSummaryDto {
  snapshotId: number;
  snapshotDate: string | null;
  headline: string | null;
  summaryText: string | null;
}

export interface PublicImpactSummaryDto {
  activeResidentsCount: number;
  safehouseCount: number;
  totalDonationsAllTime: number;
  latestPublishedSnapshots: PublicImpactSnapshotSummaryDto[];
}

export interface ResidentsBySafehouseDto {
  safehouseId: number;
  safehouseName: string | null;
  safehouseCode: string | null;
  activeResidentCount: number;
}

export interface RecentDonationDto {
  donationId: number;
  donationDate: string | null;
  donationType: string | null;
  amount: number | null;
  currencyCode: string | null;
  supporterId: number | null;
  supporterDisplayName: string | null;
}

export interface UpcomingCaseConferenceDto {
  planId: number;
  residentId: number | null;
  residentCaseNo: string | null;
  caseConferenceDate: string | null;
  planCategory: string | null;
  status: string | null;
}

export interface LatestSafehouseProgressDto {
  safehouseId: number | null;
  safehouseName: string | null;
  monthStart: string | null;
  activeResidents: number | null;
  avgEducationProgress: number | null;
  avgHealthScore: number | null;
  incidentCount: number | null;
}

export interface AdminDashboardDto {
  totalActiveResidents: number;
  activeResidentsBySafehouse: ResidentsBySafehouseDto[];
  recentDonations: RecentDonationDto[];
  upcomingCaseConferences: UpcomingCaseConferenceDto[];
  latestMonthlyProgressBySafehouse: LatestSafehouseProgressDto[];
  totalSupporters: number;
  supportersDonatedThisMonth: number;
}

export interface DonationTrendPointDto {
  year: number;
  month: number;
  totalAmount: number;
  donationCount: number;
}

export interface ReintegrationStatusCountDto {
  status: string | null;
  count: number;
}

export interface SafehousePerformanceDto {
  safehouseId: number;
  safehouseName: string | null;
  monthStart: string | null;
  activeResidents: number | null;
  avgEducationProgress: number | null;
  avgHealthScore: number | null;
  processRecordingCount: number | null;
  homeVisitationCount: number | null;
  incidentCount: number | null;
}

export interface OutcomeTrendPointDto {
  monthStart: string;
  avgEducationProgress: number | null;
  avgHealthScore: number | null;
  activeResidentsSum: number;
}
